R/zchunk_LA143.HDDCDD.R

```diff
@@ -31,65 +41,169 @@ module_energy_LA143.HDDCDD_DISABLED <- function(command, ...) {
+
+    # Reading HDDCDD files in a list so that row binding will include a filename
+    HDDCDD_data_list <- list(CDD_CCSM3x_A2 = CDD_CCSM3x_A2, CDD_CCSM3x_B1 = CDD_CCSM3x_B1,
+                             CDD_HadCM3_A2 = CDD_HadCM3_A2, CDD_HadCM3_B1 = CDD_HadCM3_B1,
+                             CDD_no_GCM_constdd = CDD_no_GCM_constdd, HDD_CCSM3x_A2 = HDD_CCSM3x_A2,
+                             HDD_CCSM3x_B1 = HDD_CCSM3x_B1, HDD_HadCM3_A2 = HDD_HadCM3_A2,
+                             HDD_HadCM3_B1 = HDD_HadCM3_B1, HDD_no_GCM_constdd = HDD_no_GCM_constdd)
+    HDDCDD_data <- bind_rows(HDDCDD_data_list, .id = 'file')
+
+    # Currently the HDDCDD data stops at 2099. If this is the case, add 2100
+    if( !"2100" %in% names( HDDCDD_data ) ){
+     HDDCDD_data <- HDDCDD_data %>% mutate(`2100` = `2099`)
+    }
+
+    # Convert data to long format and add in id variables
+    HDDCDD_data <- HDDCDD_data %>%
+      gather(year, value, -file, -country) %>%
+      mutate(
+        # Remove X years
+        year = as.integer(year),
+        # Assuming that the variable is the first three letters
+        variable = substr(file, 1, 3),
+        # Assuming that the GCM comes after "DD_" and is 6 letters
+        GCM = substr(file,5,10),
+        # Assuming that the last word is the scenario, starting at twelve letters
+        SRES = substr(file,12,length(file)),
+        # Set all negative values to 0
+        value = if_else(value < 0, 0, value)
+      )
+
+    if(OLD_DATA_SYSTEM_BEHAVIOR){
+    # Add in country iso
+    L143.HDDCDD_scen_ctry_Y <- HDDCDD_data %>%
+      # Drop file name
+      select(-file) %>%
+      # Filter only useful years
+      filter(year %in% c(HISTORICAL_YEARS, FUTURE_YEARS)) %>%
+      # Drop Cote d'Ivoire--this is a mistake in old data system
+      filter(country != "Cote d'Ivoire") %>%
+      left_join_error_no_match(GIS_ctry, by = "country")
+    } else{
+      # Add in country iso
+      L143.HDDCDD_scen_ctry_Y <- HDDCDD_data %>%
+        # Drop file name
+        select(-file) %>%
+        # Filter only useful years
+        filter(year %in% c(HISTORICAL_YEARS, FUTURE_YEARS)) %>%
+        # Remove apostrophe in Cote d'Ivoire and add in country iso by country name
+        mutate(country = if_else(country == "Cote d'Ivoire", "Cote dIvoire", country)) %>%
+        left_join_error_no_match(GIS_ctry, by = 'country')
+    }
+
+    # Serbia and Montenegro are currently combined. Copy to separated countries, assigning the same HDD and CDD to each
+    if ("scg" %in% L143.HDDCDD_scen_ctry_Y$iso){
+      # Create Serbia tibble
+      L143.HDDCDD_scen_srb_Y <- L143.HDDCDD_scen_ctry_Y %>%
+        filter(iso == "scg" ) %>%
+        mutate(iso = "srb")
+
+      # Insert Serbia tibble, change 'scg' iso to 'mne' iso
+      L143.HDDCDD_scen_ctry_Y <- L143.HDDCDD_scen_ctry_Y %>%
+        bind_rows(L143.HDDCDD_scen_srb_Y) %>%
+        mutate(iso = if_else(iso == "scg", "mne", iso))
```

Comment:
Don't you have to change iso from scg to mne before bind_rows so that you don't the original scg to mne? Don't you end up with two mne regions?